crypto/keys/pem_signer_factory_test.go

```diff
@@ -107,3 +113,101 @@ func TestPEMSignerFactoryNewSigner(t *testing.T) {
 		}
 	}
 }
+
+func TestPEMSignerFactoryGenerate(t *testing.T) {
+	for _, test := range []struct {
+		name    string
+		keygen  *keyspb.Generator
+		tree    *trillian.Tree
+		wantErr bool
+	}{
+		{
+			name: "RSA",
+			keygen: &keyspb.Generator{
+				Params: &keyspb.Generator_RsaParams{
+					RsaParams: &keyspb.Generator_RSA{},
+				},
+			},
+			tree: &trillian.Tree{
+				SignatureAlgorithm: sigpb.DigitallySigned_RSA,
+			},
+		},
+		{
+			name: "ECDSA",
+			keygen: &keyspb.Generator{
+				Params: &keyspb.Generator_EcdsaParams{
+					EcdsaParams: &keyspb.Generator_ECDSA{},
+				},
+			},
+			tree: &trillian.Tree{
+				SignatureAlgorithm: sigpb.DigitallySigned_ECDSA,
+			},
+		},
+		{
+			name: "Replace existing key",
```

Comment:
Should we be conservative and make this error? Is there a situation where we'd want this to happen?